tests/loader_phys_dev_inst_ext_tests.cpp

```diff
@@ -5246,4 +5246,155 @@ TEST(LoaderInstPhysDevExts, GetDrmDisplayEXTMixed) {
+            inst.functions->vkGetInstanceProcAddr(inst, "vkEnumeratePhysicalDeviceQueueFamilyPerformanceQueryCountersKHR"));
+    auto GetPhysicalDeviceMultisampleProperties = reinterpret_cast<PFN_vkGetPhysicalDeviceMultisamplePropertiesEXT>(
+        inst.functions->vkGetInstanceProcAddr(inst, "vkGetPhysicalDeviceMultisamplePropertiesEXT"));
+    auto GetPhysicalDeviceCalibrateableTimeDomains = reinterpret_cast<PFN_vkGetPhysicalDeviceCalibrateableTimeDomainsEXT>(
+        inst.functions->vkGetInstanceProcAddr(inst, "vkGetPhysicalDeviceCalibrateableTimeDomainsEXT"));
+    ASSERT_NE(nullptr, EnumeratePhysicalDeviceQueueFamilyPerformanceQueryCounters);
+    ASSERT_NE(nullptr, GetPhysicalDeviceMultisampleProperties);
+    ASSERT_NE(nullptr, GetPhysicalDeviceCalibrateableTimeDomains);
+
+    for (uint32_t dev = 0; dev < device_count; ++dev) {
+        uint32_t extension_count = 0;
+        std::vector<VkExtensionProperties> device_extensions;
+        bool supports_query = false;
+        bool supports_samples = false;
+        bool supports_timestamps = false;
+        ASSERT_EQ(VK_SUCCESS,
+                  inst->vkEnumerateDeviceExtensionProperties(physical_devices[dev], nullptr, &extension_count, nullptr));
+        ASSERT_GT(extension_count, 0);
+        device_extensions.resize(extension_count);
+        ASSERT_EQ(VK_SUCCESS, inst->vkEnumerateDeviceExtensionProperties(physical_devices[dev], nullptr, &extension_count,
+                                                                         device_extensions.data()));
+        for (uint32_t ext = 0; ext < extension_count; ++ext) {
+            if (string_eq(VK_KHR_PERFORMANCE_QUERY_EXTENSION_NAME, &device_extensions[ext].extensionName[0])) {
+                supports_query = true;
+            }
+            if (string_eq(VK_EXT_SAMPLE_LOCATIONS_EXTENSION_NAME, &device_extensions[ext].extensionName[0])) {
+                supports_samples = true;
+            }
+            if (string_eq(VK_EXT_CALIBRATED_TIMESTAMPS_EXTENSION_NAME, &device_extensions[ext].extensionName[0])) {
+                supports_timestamps = true;
+            }
+        }
+
+        // For physical device extensions, they should work for devices that support it and crash for those that don't.
+        if (supports_query) {
+            ASSERT_EQ(VK_SUCCESS, EnumeratePhysicalDeviceQueueFamilyPerformanceQueryCounters(physical_devices[dev], 0, nullptr,
+                                                                                             nullptr, nullptr));
+        } else {
+            ASSERT_DEATH(
+                EnumeratePhysicalDeviceQueueFamilyPerformanceQueryCounters(physical_devices[dev], 0, nullptr, nullptr, nullptr),
+                "");
+            ASSERT_FALSE(
+                log.find("ICD associated with VkPhysicalDevice does not support "
+                         "EnumeratePhysicalDeviceQueueFamilyPerformanceQueryCountersKHR"));
```

Comment:
I don't think this code will ever be run. Death tests presume that the test ends there from my understanding.

This also implies that the test never gets past the the first device, so its worth double checking.

Reply:
Nope, death test assumes that function asserts  It continues on and actually does hit these checks.  The messages I'm checking for are actually output by the loader.

Reply:
How convenient! Well then I don't have to second guess all the death tests!